lib/experimental/class-gutenberg-rest-global-styles-revisions-controller.php

```diff
@@ -0,0 +1,363 @@
+<?php
+/**
+ * REST API: Gutenberg_REST_Global_Styles_Revisions_Controller class, inspired by WP_REST_Revisions_Controller.
+ *
+ * @package WordPress
+ * @subpackage REST_API
+ * @since 6.3.0
+ */
+
+/**
+ * Core class used to access global styles revisions via the REST API.
+ *
+ * @since 6.3.0
+ *
+ * @see WP_REST_Controller
+ */
+class Gutenberg_REST_Global_Styles_Revisions_Controller extends WP_REST_Controller {
+	/**
+	 * Parent post type.
+	 *
+	 * @since 6.3.0
+	 * @var string
+	 */
+	private $parent_post_type;
+
+	/**
+	 * The base of the parent controller's route.
+	 *
+	 * @since 6.3.0
+	 * @var string
+	 */
+	private $parent_base;
+
+	/**
+	 * Constructor.
+	 *
+	 * @since 6.3.0
+	 */
+	public function __construct() {
+		$this->parent_post_type = 'wp_global_styles';
+		$this->rest_base        = 'revisions';
+		$this->parent_base      = 'global-styles';
+		$this->namespace        = 'wp/v2';
+	}
+
+	/**
+	 * Registers the controllers routes.
+	 *
+	 * @return void
+	 */
+	public function register_routes() {
+		register_rest_route(
```

Comment:
Will there also be a route for accessing an individual revision / deleting a revision?

I imagine that's not required for the initial version of the endpoint (exposing a list of revisions in the site editor UI), but could be a good follow-up PR to ensure consistency?

Reply:
I agree. I expect one day they'll both be needed. I haven't heard anything about it for 6.3.